fdbserver/ptxn/test/TestTLogServer.actor.cpp

```diff
@@ -636,11 +636,73 @@ TEST_CASE("/fdbserver/ptxn/test/lock_tlog") {
 	return Void();
 }
 
+ACTOR Future<std::pair<std::vector<Standalone<StringRef>>, std::vector<Version> >> commitInjectReturnVersions(std::shared_ptr<ptxn::test::TestDriverContext> pContext,
+                                                              ptxn::StorageTeamID storageTeamID,
+                                                              int numCommits) {
+	state ptxn::test::print::PrintTiming printTiming("tlog/commitInject");
+
+	state const ptxn::TLogGroupID tLogGroupID = pContext->storageTeamIDTLogGroupIDMapper.at(storageTeamID);
+	state std::shared_ptr<ptxn::TLogInterfaceBase> pInterface = pContext->getTLogLeaderByStorageTeamID(storageTeamID);
+	ASSERT(pInterface);
+
+	state Version currVersion = 0;
+	state Version prevVersion = currVersion;
+	increaseVersion(currVersion);
+
+	state std::vector<ptxn::TLogCommitRequest> requests;
+	state std::vector<Standalone<StringRef>> writtenMessages;
+	state std::vector<Version> versions;
+	for (auto i = 0; i < numCommits; ++i) {
+		generateMutations(currVersion, 16, { storageTeamID }, pContext->commitRecord);
+		auto serialized = serializeMutations(currVersion, storageTeamID, pContext->commitRecord);
+		std::unordered_map<ptxn::StorageTeamID, StringRef> messages = { { storageTeamID, serialized } };
+		requests.emplace_back(ptxn::test::randomUID(),
+		                      pContext->storageTeamIDTLogGroupIDMapper[storageTeamID],
+		                      serialized.arena(),
+		                      messages,
+		                      prevVersion,
+		                      currVersion,
+		                      0,
+		                      0,
+		                      Optional<UID>());
+		writtenMessages.emplace_back(getLogEntryContent(requests.back(), pInterface->id()));
+		versions.push_back(currVersion);
+		prevVersion = currVersion;
+		increaseVersion(currVersion);
+	}
+	printTiming << "Generated " << numCommits << " commit requests" << std::endl;
+	{
+		std::mt19937 g(deterministicRandom()->randomUInt32());
+		std::shuffle(std::begin(requests), std::end(requests), g);
+	}
+
+	state std::vector<Future<ptxn::TLogCommitReply>> replies;
+	state int index = 0;
+	for (index = 0; index < numCommits; ++index) {
+		printTiming << "Sending version " << requests[index].version << std::endl;
+		replies.push_back(pInterface->commit.getReply(requests[index]));
+		wait(delay(0.5));
+	}
+	wait(waitForAll(replies));
+	printTiming << "Received all replies" << std::endl;
+
+	return std::make_pair(writtenMessages, versions);
+}
+
+Key persistStorageTeamMessageRefsKey(UID id, ptxn::StorageTeamID storageTeamId, Version version) {
```

Comment:
Looks like this function is duplicated. Consider removing this one.

Reply:
it is because the `static `you mentioned in the first comment, will discuss offline.